integreat_cms/api/v3/pages.py

```diff
@@ -20,25 +20,34 @@ def transform_page(page_translation):
     :param page_translation: single page translation object
     :type page_translation: ~integreat_cms.cms.models.pages.page_translation.PageTranslation
 
+    :raises ~django.http.Http404: HTTP status 404 if a parent is archived
+
     :return: data necessary for API
     :rtype: dict
     """
+    fallback_parent = {
+        "id": 0,
+        "url": None,
+        "path": None,
+    }
+
     parent_page = page_translation.page.parent
-    if parent_page:
-        parent_absolute_url = parent_page.get_public_translation(
+    if parent_page and not parent_page.explicitly_archived:
+        parent_public_translation = parent_page.get_public_translation(
             page_translation.language.slug
-        ).get_absolute_url()
-        parent = {
-            "id": parent_page.id,
-            "url": settings.BASE_URL + parent_absolute_url,
-            "path": parent_absolute_url,
-        }
+        )
+        if parent_public_translation:
+            parent_absolute_url = parent_public_translation.get_absolute_url()
+            parent = {
+                "id": parent_page.id,
+                "url": settings.BASE_URL + parent_absolute_url,
+                "path": parent_absolute_url,
+            }
+        else:
+            parent = fallback_parent
```

Comment:
If I understood correctly, we already circumvent this problem earlier by not calling this method for pages without public parents, right? So just throwing an exception without handling it could also be fine.

Reply:
Indeed. I think I had to implement this before the cache_tree already filtered by publicly visible content. And we don't want 404s in the normal pages endpoint ;-) Fixed now.